assignments/prototypes.js

```diff
@@ -102,9 +136,94 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+
 
   // Stretch task: 
   // * Create Villian and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villians different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villian and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villian and one a hero and fight it out with methods!
+
+
+  function Villain(villainAttributes){
+    Humanoid.call(this, villainAttributes);
+    this.attackValue = villainAttributes.attackValue;
+  }
+
+  function Hero(heroAttributes){
+    Humanoid.call(this, heroAttributes);
+    this.attackValue = heroAttributes.attackValue;
+  }
+
+  Villain.prototype = Object.create(Humanoid.prototype);
+  Hero.prototype = Object.create(Humanoid.prototype);
+
+  Villain.prototype.attack = function(hero){
+    if(hero.hp > this.attackValue){
+      return hero.hp -= this.attackValue;
+    }else if(hero.hp <= this.attackValue){
+      return hero.destroy();
+    }
+  }
+  Hero.prototype.attack = function(villain){
+    if(villain.hp > this.attackValue){
+      return villain.hp -= this.attackValue;
+    }else if(villain.hp <= this.attackValue){
+      return villain.destroy();
+    }
```

Comment:
You could forego the `else if` here since it encompasses everything that the first `if` doesn't:
```javascript
Hero.prototype.attack = function(villain) {
    if (villain.hp > this.attackValue){
      return villain.hp -= this.attackValue;
    } else {
      return villain.destroy();
    }
}
```